### PR TITLE
Feat: Testing and Linting

### PR DESCRIPTION
**Pull request by CodeWithEmad**
Assorted testing and linting improvements:

* Makefile with test and lint targets
* Long lines reformated with black
* Sort imports with isort

In addition:
* All configs moved to 1 place to better support the unification of plugins
* Typing added for various functions

PS. Looks like Actions are not enabled on this repository.

### CODE REVIEW

**Comment by CodeWithEmad** [comment]
@regisb we talked earlier about removing `.pre-commit-config.yaml`, `.editorconfig`, and `linter.sh` since these are not a part of the cookie-cutter. should I remove them as a part of cleanup in this PR?

**Comment by Faraz32123** [comment]
Hi @CodeWithEmad, thanks for creating a PR. I will look into it.
Can u please resolve the merge conflicts?

**Comment by regisb** [comment]
@CodeWithEmad yes I think we should get rid of these files.

**Comment by CodeWithEmad** [comment]
and what about `.gitlab-ci.yml`?

**Comment by regisb** [comment]
We very much need that one! That's because our private CI runs on Gitlab.

**Comment by CodeWithEmad** [comment]
Oh, that's good to know!

File: tutorcredentials/__about__.py [hunk 1]
```diff
@@ -1,2 +1 @@
 __version__ = "16.0.3"
-
```
Comment (Faraz32123) [inline tutorcredentials/__about__.py:2]: why removing a newline? It's a python file.

File: tutorcredentials/plugin.py [hunk 2]
```diff
@@ -19,78 +19,67 @@
...
+        "LOGO_WHITE_URL_PNG": "{{ CREDENTIALS_LOGO_URL }}",
+        "LOGO_WHITE_URL_SVG": "",
+        "MYSQL_DATABASE": "credentials",
+        "MYSQL_USERNAME": "credentials",
+        "OAUTH2_KEY": "credentials-key",
+        "OAUTH2_KEY_DEV": "credentials-key-dev",
+        "OAUTH2_KEY_SSO": "credentials-key-sso",
+        "OAUTH2_KEY_SSO_DEV": "credentials-key-sso-dev",
+        "PLATFORM_NAME": "{{ PLATFORM_NAME }}",
+        "PRIVACY_POLICY_URL": "{{ LMS_HOST }}/privacy-policy",
```
Comment (Faraz32123) [inline tutorcredentials/plugin.py:47]: `"{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}/privacy-policy"`

File: tutorcredentials/plugin.py [hunk 3]
```diff
@@ -19,78 +19,67 @@
...
+        "SITE_NAME": "{{ LMS_HOST }}",
+        "SOCIAL_AUTH_REDIRECT_IS_HTTPS": False,
+        "SOCIAL_AUTH_EDX_OAUTH2_ISSUER": "{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}",
+        "SOCIAL_AUTH_EDX_OAUTH2_URL_ROOT": "http://lms:8000",
+        "SOCIAL_AUTH_EDX_OAUTH2_KEY": "credentials-sso-key",
+        "SOCIAL_AUTH_EDX_OAUTH2_LOGOUT_URL": "{{ LMS_HOST }}/logout",
```
Comment (Faraz32123) [inline tutorcredentials/plugin.py:53]: `"{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}/logout"`

File: tutorcredentials/plugin.py [hunk 4]
```diff
@@ -19,78 +19,67 @@
...
+        "THEME_NAME": "edx-theme",
+        "TOS_URL": "{{ LMS_HOST }}/tos",
```
Comment (Faraz32123) [inline tutorcredentials/plugin.py:55]: `"{% if ENABLE_HTTPS %}https{% else %}http{% endif %}://{{ LMS_HOST }}/tos"`

File: tutorcredentials/plugin.py [hunk 5]
```diff
@@ -19,78 +19,67 @@
...
+    },
+    "unique": {
+        "MYSQL_PASSWORD": "{{ 8|random_string }}",
+        "SOCIAL_AUTH_EDX_OAUTH2_SECRET": "{{ 16|random_string }}",
+        "BACKEND_SERVICE_EDX_OAUTH2_SECRET": "{{ 16|random_string }}",
+        "OAUTH2_SECRET": "{{ 16|random_string }}",
+        "OAUTH2_SECRET_DEV": "{{ 16|random_string }}",
+        "OAUTH2_SECRET_SSO": "{{ 16|random_string }}",
+        "OAUTH2_SECRET_SSO_DEV": "{{ 16|random_string }}",
+    },
```
Comment (Faraz32123) [inline tutorcredentials/plugin.py:65]: please add an empty key with name `overrides` with a comment in it.
```
"overrides" : {
        # Add values to override settings from Tutor core or other plugins here.
        # Each override is a pair, (setting_name, new_value). For example:
        # ("PLATFORM_NAME", "My platform"),
}
```

Comment (Faraz32123) [reply tutorcredentials/plugin.py:65]: It's a choice to add it or not to add it.

**Review by Faraz32123 — COMMENTED** [review]
LGTM! with some changes requested.

File: tutorcredentials/plugin.py (hunk 2 above)
Comment (regisb) [reply tutorcredentials/plugin.py:47]: I think that this PR needs to be rebased on top of master. @CodeWithEmad you seem to be missing some of the latest changes. Sorry!

Comment (Faraz32123) [reply tutorcredentials/plugin.py:47]: Yes @regisb, you are right.

File: tutorcredentials/plugin.py (hunk 5 above)
Comment (CodeWithEmad) [reply tutorcredentials/plugin.py:65]: Actually, no other MFE has this empty `overrides` key, so I think it's better to leave it this way.

File: tutorcredentials/plugin.py (hunk 2 above)
Comment (CodeWithEmad) [reply tutorcredentials/plugin.py:47]: Oh, I missed that one. 

> I think that this PR needs to be rebased on top of master

It's Done.

File: tutorcredentials/__about__.py (hunk 1 above)
Comment (CodeWithEmad) [reply tutorcredentials/__about__.py:2]: I know that! :) 
there were 2 blank lines at the end of this file. 
Screenshot from 2023-11-23 21-01-40

GitHub doesn't show the `1 blank line` at the end of the files and only warns you if there are not any.

File: tutorcredentials/plugin.py (hunk 5 above)
Comment (Faraz32123) [reply tutorcredentials/plugin.py:65]: yes, it was a choice anyway.

File: tutorcredentials/plugin.py (hunk 2 above)
Comment (Faraz32123) [reply tutorcredentials/plugin.py]: Actually `PRIVACY_POLICY_URL` was removed in the latest code of main branch.
Can u please rebase it again or sync it with master as `PRIVACY_POLICY_URL` is still in the PR.
Thanks for bearing it with me 👍

Comment (CodeWithEmad) [reply tutorcredentials/plugin.py]: It's Done.

**Review by Faraz32123 — APPROVED** [review]
LGTM!

File: tutorcredentials/plugin.py [hunk 6]
```diff
@@ -19,78 +19,66 @@
...
-        ("CREDENTIALS_PLATFORM_NAME", "{{ PLATFORM_NAME }}"),
-        ("CREDENTIALS_SITE_NAME", "{{ LMS_HOST }}"),
-        ("CREDENTIALS_SOCIAL_AUTH_REDIRECT_IS_HTTPS", False),
-        ("CREDENTIALS_SOCIAL_AUTH_EDX_OAUTH2_ISSUER", "https://{{ LMS_HOST }}"),
-        ("CREDENTIALS_SOCIAL_AUTH_EDX_OAUTH2_URL_ROOT", "http://lms:8000"),
-        ("CREDENTIALS_SOCIAL_AUTH_EDX_OAUTH2_KEY", "credentials-sso-key"),
-        ("CREDENTIALS_SOCIAL_AUTH_EDX_OAUTH2_LOGOUT_URL", "{{ LMS_HOST }}/logout"),
-        ("CREDENTIALS_THEME_NAME", "edx-theme"),
-        ("CREDENTIALS_REPOSITORY", "https://github.com/openedx/credentials.git"),
-        ("CREDENTIALS_REPOSITORY_VERSION", "{{ OPENEDX_COMMON_VERSION }}"),
```
Comment (regisb) [inline tutorcredentials/plugin.py:58]: @CodeWithEmad you missed these two settings: CREDENTIALS_REPOSITORY, CREDENTIALS_REPOSITORY_VERSION

As a consequence, CI fails during `tutor config save`. Can you please fix this? cc @Faraz32123

Comment (Faraz32123) [reply tutorcredentials/plugin.py:58]: I'll fix it asap.

Comment (Faraz32123) [reply tutorcredentials/plugin.py:58]: fixed it.

**Pull request merged**